### PR TITLE
Tests refactoring

### PR DESCRIPTION
**Pull request by CDimonaco**
# Description

As title.

We refactor the tests, to get rid of the Routes module in the controller tests, and get a correct usages of factories, specially in multi insert statements.

## How was this tested?

Automated tests

### CODE REVIEW

File: test/trento/application/usecases/sap_systems_test.exs [hunk 1]
```diff
@@ -23,18 +23,16 @@ defmodule Trento.SapSystemsTest do
       } = insert(:sap_system)
 
       application_instances =
-        0..4
-        |> Enum.map(fn _ ->
-          insert(:application_instance_without_host, sap_system_id: sap_system_id)
-        end)
-        |> Enum.sort_by(&{&1.instance_number, &1.host_id})
+        Enum.sort_by(
```
Comment (fabriziosestito) [inline test/trento/application/usecases/sap_systems_test.exs:26]: keep in mind you can pipe here ````elixir 5 |> insert_list(...) |> Enum.sort(...```

File: test/trento/application/integration/discovery/discovery_test.exs [hunk 2]
```diff
@@ -83,28 +83,26 @@ defmodule Trento.Integration.DiscoveryTest do
...
-        agent_id: Faker.UUID.v4(),
-        discovery_type: "discovery_type",
-        payload: %{},
-        inserted_at: Timex.shift(DateTime.utc_now(), days: -11)
-      )
-    end
+    insert_list(
+      10,
+      :discovery_event,
+      agent_id: Faker.UUID.v4(),
```
Comment (arbulu89) [inline test/trento/application/integration/discovery/discovery_test.exs:89]: We can safely remove this `agent_id` and `payload` entries. It is generated by the factory

**Review by fabriziosestito — APPROVED** [review]
Great!

File: test/trento/application/integration/discovery/discovery_test.exs [hunk 3]
```diff
@@ -83,28 +83,26 @@ defmodule Trento.Integration.DiscoveryTest do
...
-      insert(
-        :discarded_discovery_event,
-        payload: %{},
-        inserted_at: Timex.shift(DateTime.utc_now(), days: -11)
-      )
-    end
+    insert_list(
+      10,
+      :discarded_discovery_event,
+      payload: %{},
```
Comment (arbulu89) [inline test/trento/application/integration/discovery/discovery_test.exs:103]: We can remove the `payload` field

**Review by arbulu89 — APPROVED** [review]
Go go!

**Pull request merged**